src/execute-query.ts

```diff
@@ -0,0 +1,43 @@
+import { ConnectionConfig } from 'mysql';
+import * as serverlessMysql from 'serverless-mysql';
+
+const defaultConfig: ConnectionConfig = {
+    host: process.env.host,
+    user: process.env.user,
+    password: process.env.password,
+    database: process.env.database,
+};
+const mysql = serverlessMysql({ config: defaultConfig });
+
+async function executeQuery<T>(query: string, dbConfig: ConnectionConfig): Promise<{ data: T; error: string }> {
```

Comment:
❓ TypeScript noob question... `T` is the same as `Type` right? is it just shorthand?

Reply:
It indicates a Generic. It is sort of like a variable name. ie:

const queryResponse = executeQuery<string[]>('select name from table');

would return a type of { error: string, data: string[] }

T is a common syntax for generics when you only have one Type. This can be renamed to Type if we feel like it is easier to understand. But the word Type has a greater chance of name clashing than T.